task2/rest-api-basics/repository/src/main/java/com/epam/esm/repository/impl/GiftCertificateRepositoryImpl.java

```diff
@@ -0,0 +1,321 @@
+            "WHERE t.name = ?";
+    private static final String FIND_ALL_WITH_TAGS_BY_PART_OF_NAME_OR_DESCRIPTION_QUERY =
+            "SELECT gc.id gc_id, gc.name gc_name, gc.description, gc.price, gc.duration, gc.create_date, gc.last_update_date, t.id t_id, t.name t_name " +
+            "FROM gift_certificate gc " +
+            "JOIN gift_certificate_tag gct ON gc.id = gct.gift_certificate_id " +
+            "JOIN tag t ON gct.tag_id = t.id " +
+            "WHERE gc.name LIKE ? OR gc.description LIKE ?";
+    private static final String FIND_ALL_WITH_TAGS_BY_TAG_NAME_AND_PART_OF_NAME_OR_DESCRIPTION_QUERY =
+            "SELECT gc.id gc_id, gc.name gc_name, gc.description, gc.price, gc.duration, gc.create_date, gc.last_update_date, t.id t_id, t.name t_name " +
+            "FROM gift_certificate gc " +
+            "JOIN gift_certificate_tag gct ON gc.id = gct.gift_certificate_id " +
+            "JOIN tag t ON gct.tag_id = t.id " +
+            "WHERE t.name = ? AND (gc.name LIKE ? OR gc.description LIKE ?)";
+    private static final String FIND_BY_ID_QUERY =
+            "SELECT id gc_id, name gc_name, description, price, duration, create_date, last_update_date " +
+            "FROM gift_certificate " +
+            "WHERE id = ?";
+    private static final String FIND_BY_NAME_QUERY =
+            "SELECT id gc_id, name gc_name, description, price, duration, create_date, last_update_date " +
+            "FROM gift_certificate " +
+            "WHERE name = ?";
+    private static final String FIND_BY_TAG_NAME_QUERY =
+            "SELECT gc.id gc_id, gc.name gc_name, gc.description, gc.price, gc.duration, gc.create_date, gc.last_update_date " +
+            "FROM gift_certificate gc " +
+            "JOIN gift_certificate_tag gct ON gc.id = gct.gift_certificate_id " +
+            "JOIN tag t ON gct.tag_id = t.id " +
+            "WHERE t.name = ?";
+    private static final String SAVE_QUERY =
+            "INSERT INTO gift_certificate (name, description, price, duration, create_date) " +
+            "VALUES (?, ?, ?, ?, ?)";
+    private static final String SAVE_FK_GC_TAG_QUERY =
+            "INSERT INTO gift_certificate_tag (gift_certificate_id, tag_id) VALUES (?, ?)";
+    private static final String DELETE_BY_NAME_QUERY =
+            "DELETE FROM gift_certificate WHERE name = ?";
+    private static final String UPDATE_BY_ID_QUERY =
+            "UPDATE gift_certificate SET " +
+            "name = COALESCE(?, name), " +
+            "description = COALESCE(?, description), " +
+            "price = COALESCE(?, price), " +
+            "duration = COALESCE(?, duration), " +
+            "last_update_date = ? " +
+            "WHERE id = ?";
+
+    @Override
+    public GiftCertificate findById(Long id) throws RepositoryException {
+        try {
+            return jdbcTemplate.queryForObject(FIND_BY_ID_QUERY, new GiftCertificateMapper(), id);
+        } catch (EmptyResultDataAccessException e) {
+            return null;
```

Comment:
@igor-zhuravlev What does it mean? You can apply ifexist method end return exception with message

Reply:
It throws anyway if not found, thats why I had used catch block